config/application.rb

```diff
@@ -87,5 +90,9 @@ class Application < Rails::Application
       end
 
     end
+
+    config.after_initialize do
```

Comment:
Running a settings validator after all initializers have been run is probably too late for most stuff/fruitless art. The initializers usually need to know, how they should initialize aka configure XY.